### PR TITLE
Add jasmine CLI test runner

### PR DESCRIPTION
**Pull request by viddo**
To make spec bundle smaller and have quicker feedback on tests that doesn’t require a browser window/DOM.

Basically same changes that was done on cartodb repo, https://github.com/CartoDB/cartodb/pull/7997 

@alonsogarciapablo ok?

### CODE REVIEW

File: grunt/tasks/watch.js [hunk 1]
```diff
@@ -3,8 +3,20 @@
  *
  */
 module.exports = {
-  task: function() {
+  task: function () {
     return {
+      jasmine: {
```
Comment (alonsogarciapablo) [inline grunt/tasks/watch.js]: This is great to get some instant feedback when writing/changing tests, but I personally find it not so useful when doing something else (eg: changing tests that are not executed by the `npm-test` task). It makes things a bit slower (even if tests are very fast). I see there's also `npm run test-watch`, so perhaps we can remove this?

Comment (viddo) [inline grunt/tasks/watch.js]: Sure, I'll 🔥 it then

**Comment by alonsogarciapablo** [comment]
Thanks for opening this PR @viddo! 

The only thing I don't like much about this is that we end having `test/spec` and `spec/specs` which might be a bit confusing... Perhaps we can have `spec/unit` and `spec/views` or something along those lines?

**Comment by viddo** [comment]
Agree, although the (remaining) tests in `test/spec` dir are not only for views per se, so I'd leave that dir name as it is.

**Comment by alonsogarciapablo** [comment]
What directory structure are you suggesting then?

**Comment by viddo** [comment]
`spec/unit` (as you proposed, already changed), and leave test/spec as it is.

**Comment by alonsogarciapablo** [comment]
> spec/unit (as you proposed, already changed), and leave test/spec as it is.

I was actually suggesting having all tests under one main folder and then having a subfolder for "unit" vs "browser" tests, so that anyone coming to CartoDB.js will surely know where to add new tests...

**Comment by viddo** [comment]
Ah, yes, I misunderstood then. Agree, updated again 😄

**Comment by alonsogarciapablo** [comment]
👍 Thanks!

**Comment by viddo** [comment]
Good to go?

**Comment by alonsogarciapablo** [comment]
:shipit:

**Pull request merged**